packages/client/src/lib.js

```diff
@@ -193,6 +193,26 @@ class Web3Storage {
     throw Error('.delete not implemented yet')
   }
 
+  /**
+   * @param {Service} service
+   * @param {CIDString} cid
+   * @returns {Promise<CIDString>}
+   */
+  static async status ({ endpoint, token }, cid) {
+    const url = new URL(`/status/${cid}`, endpoint)
+    const res = await fetch(url, {
+      method: 'GET',
+      headers: Web3Storage.headers(token)
+    })
+    if (res.status === 404) {
+      return undefined
+    }
+    if (!res.ok) {
+      throw new Error(res.statusText)
+    }
+    return res.json()
```

Comment:
Shall we transform the date strings to `Date` objects? Asking because we do in nft.storage.

Reply:
i'm disinclined to do that. The user might want moments, or luxons or date-fns, or might not care about those fields. An ISO string is a nice thing.